cpg_workflows/stages/gcnv.py

```diff
@@ -0,0 +1,76 @@
+"""
+Stages that run gCNV.
+"""
+
+from cpg_utils.config import get_config
+from cpg_workflows.jobs import gatk
+from cpg_workflows.workflow import Sample, stage, StageInput, StageOutput, Dataset, DatasetStage, SampleStage
+from cpg_workflows.stages.align import Align
+from .. import get_batch
+
+
+@stage(required_stages=Align)
+class PrepareIntervals(DatasetStage):
+    def _intervals_file(self, dataset):
+        h = dataset.alignment_inputs_hash()
+        return dataset.prefix() / 'gcnv' / h / f'{dataset.name}.interval_list'
+
+    def expected_outputs(self, dataset):
+        return {'intervals': self._intervals_file(dataset)}
```

Comment:
A couple of thoughts here... Happy to discuss on all accounts.

* The Dataset Stage will always run once per Dataset, so splitting off the intervals file generation into a private method with a dataset as an argument feels a little redundant. The logic is sound but I don't think it provides an explicit benefit.

The prefix setting is interesting for 2 reasons:

1. The alignment hash for the workflow can be overridden (see [here](https://github.com/populationgenomics/production-pipelines/blob/main/cpg_workflows/workflow.py#L857-L884)), but by explicitly calculating it you'd potentially be overriding the override.
2. Instead of building on `dataset.prefix()` have a look at `self.prefix`:
    - returns `workflow.prefix` / name (Stage name, so `PrepareIntervals`) 
    - Workflow prefix is:
        - bucket the workflow is running under (e.g. if you're running across multiple projects as a higher level dataset, it will be that higher dataset bucket. If you're running the workflow for one dataset, this will == that project bucket)
        - workflow name (here `gcnv`)
        - alignments hash for the workflow (can be set explicitly)
    - example: `gs://cpg-broad-rgp-test/gatk_sv/b1b0be5754cfebd117b2a0e50b2d4087fe1af6_100/ClusterBatch/<files>`